code/controllers/subsystem/codex.dm

```diff
@@ -168,3 +241,120 @@ SUBSYSTEM_DEF(codex)
+		if(db_serial == CODEX_SERIAL_ALWAYS_VALID)
+			to_chat(world, span_debug("Codex: Special Database Serial detected. Data may be inaccurate or out of date."))
+		else
+			to_chat(world, span_debug("Codex: Database out of date, Rebuilding..."))
+			prepare_search_database(TRUE) //recursiveness funny,,
+			return
+
+	if(drop_existing)
+		to_chat(world, span_debug("Codex: Collation complete.\nCodex: Index ready."))
+		return
+	to_chat(world, span_debug("Codex: Database Serial validated.\nCodex: Loading complete."))
+
+/datum/controller/subsystem/codex/proc/create_db()
+	// No index? Make one.
+
+	to_chat(world, span_debug("Codex: Writing new database file..."))
+	//We explicitly store the DB in the root directory, so that TGS builds wipe it.
+	codex_index = new(CODEX_SEARCH_INDEX_FILE)
+
+	//Create the initial schema
+	var/database/query/init_cursor = new({"
+/* Holds the revision the index was compiled for. If it's different then live, we need to regenerate the index.*/
+	CREATE TABLE "_info" (
+		"revision"	TEXT
+	);
+	"})
+
+	if(!init_cursor.Execute(codex_index))
+		to_chat(world, span_debug("Codex: ABORTING! Database error: [init_cursor.Error()] | [init_cursor.ErrorMsg()]"))
+		return
+
+
+	init_cursor.Add({"
+/* Holds all codex entries to enable accelerated text search.*/
+	CREATE TABLE "codex_entries" (
+		"name"	TEXT NOT NULL,
+		"lore_text"	TEXT,
+		"mechanics_text"	TEXT,
+		"antag_text"	TEXT,
+		PRIMARY KEY("name")
+		);
+	"})
+	if(!init_cursor.Execute(codex_index))
+		to_chat(world, span_debug("Codex: ABORTING! Database error: [init_cursor.Error()] | [init_cursor.ErrorMsg()]"))
+		return
+
+	var/revid = GLOB.revdata.commit
+	if(!revid) //zip download, you're on your own pissboy, The index will always be considered valid.
+		revid = CODEX_SERIAL_ALWAYS_VALID
+
+	//Insert the revision header.
+	init_cursor.Add("INSERT INTO _info (revision) VALUES (?)", revid)
+	if(!init_cursor.Execute(codex_index))
+		to_chat(world, span_debug("Codex: ABORTING! Database error: [init_cursor.Error()] | [init_cursor.ErrorMsg()]"))
+		return
+
+/datum/controller/subsystem/codex/proc/build_db_index()
+	to_chat(world, span_debug("Codex: Building search index."))
+
+	var/database/query/cursor = new
+	var/total_entries = length(all_entries)
+	to_chat(world, span_debug("\tCodex: Collating [total_entries] records..."))
+	var/record_id = 0 //Counter for debugging.
+	for(var/datum/codex_entry/entry as anything in all_entries)
+		cursor.Add(
+			"INSERT INTO codex_entries (name, lore_text, mechanics_text, antag_text) VALUES (?,?,?,?)",
+			entry.name,
+			entry.lore_text,
+			entry.mechanics_text,
+			entry.antag_text
+		)
+		if(!cursor.Execute(codex_index))
+			to_chat(world, span_debug("Codex: ABORTING! Database error: [cursor.Error()] | [cursor.ErrorMsg()]"))
+			return
+		record_id++
+		if((!(record_id % 100)) || (record_id == total_entries))
+			to_chat(world, span_debug("\tCodex: [record_id]/[total_entries]..."))
+		CHECK_TICK //We'd deadlock the server otherwise.
```

Comment:
```suggestion
	for(var/datum/codex_entry/entry as anything in all_entries)
		cursor.Add(
			"INSERT INTO codex_entries (name, lore_text, mechanics_text, antag_text) VALUES (?,?,?,?)",
			entry.name,
			entry.lore_text,
			entry.mechanics_text,
			entry.antag_text
		)

		if(!cursor.Execute(codex_index))
			to_chat(world, span_debug("Codex: ABORTING! Database error: [cursor.Error()] | [cursor.ErrorMsg()]"))
			return

		record_id++
		if((!(record_id % 100)) || (record_id == total_entries))
			to_chat(world, span_debug("\tCodex: [record_id]/[total_entries]..."))

		CHECK_TICK //We'd deadlock the server otherwise.
```